scripts/blog_template.py

```diff
@@ -40,27 +74,48 @@ def fix_img_paths(data: pd.DataFrame) -> pd.Series:
     A function for fixing image paths generated by Microsoft forms
     """
 
-    assert 'image_file' in data.columns 
+    assert "image_file" in data.columns
+
+    return (
+        data["image_file"]
+        .fillna("")
+        .apply(
+            lambda x: ",".join(
+                [x.split("/")[-1].replace("%20", "_") for x in x.split(";")]
+            )
+        )
+    )
 
-    return data['image_file'].fillna("").apply(lambda x: x.split("/")[-1].replace("%20","_"))
 
-def main(data_file, output_path):
+def main(data_file: str, output_path: str, date: str) -> None:
     """
     Main script for loading data in pandas iterating over rows and passing it to render_md
     """
     # open data file from csv
     # expects csv file containing windows encoding from forms
-    working_file = pd.read_csv(data_file)
+    working_file = pd.read_csv(data_file, encoding="latin")
 
-    working_file['image_file'] = fix_img_paths(working_file)
+    working_file = column_mapper(working_file)
+
+    working_file["image_file"] = fix_img_paths(working_file)
+
+    # randomly shuffle the rows
+    working_file = working_file.sample(frac=1, random_state=42).reset_index(drop=True)
+
+    # create a column with publish date
+    # takes a YYYY-MM-DD string
+    # periods is the number of rows
+    # freq is D for days
+    working_file["Publish_date"] = pd.date_range(
+        date, periods=working_file.shape[0], freq="D"
+    ).astype(str)
```

Comment:
To create a "Publish_date" column we use pandas convenient date_range function. 
We take the passed date (string in format YYYY-MM-DD), a number of periods, that corresponds to the number of rows in our dataframe, and set frequency to "D" for days to get a date range for each day we have a post from a start date.

We coerce the type here to string because `pd.date_range` returns a Timestamp type by default